src/main.rs

```diff
@@ -285,22 +283,22 @@ impl Config {
         result
     }
 
-    fn output_processing_mode(&self) -> OutputProcessingMode {
+    fn regress_on(&self) -> RegressOn {
         match self.args.regress.as_str() {
-            "error" => OutputProcessingMode::RegressOnErrorStatus,
-            "non-error" => OutputProcessingMode::RegressOnNonCleanError,
-            "ice" => OutputProcessingMode::RegressOnIceAlone,
-            "non-ice" => OutputProcessingMode::RegressOnNotIce,
-            "success" => OutputProcessingMode::RegressOnSuccessStatus,
+            "error" => RegressOn::ErrorStatus,
+            "non-error" => RegressOn::NonCleanError,
+            "ice" => RegressOn::IceAlone,
+            "non-ice" => RegressOn::NotIce,
+            "success" => RegressOn::SuccessStatus,
             setting => panic!("Unknown --regress setting: {:?}", setting),
         }
     }
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, Debug, StructOpt)]
 /// Customize what is treated as regression.
-enum OutputProcessingMode {
-    /// `RegressOnErrorStatus`: Marks test outcome as `Regressed` if and only if
+enum RegressOn {
```

Comment:
This is quiet a "big one":

all variants have the same prefix: `RegressOn`
I decided to rename `OutputProcessingMode` to `RegressOn` because I thought it better fits the needings here.